tests/src/test_helpers/ssl_helpers.c

```diff
@@ -685,9 +685,24 @@ int mbedtls_test_ssl_endpoint_certificate_init(mbedtls_test_ssl_endpoint *ep,
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     if (opaque_alg != 0) {
-        TEST_EQUAL(mbedtls_pk_wrap_as_opaque(cert->pkey, &key_slot,
-                                             opaque_alg, opaque_usage,
-                                             opaque_alg2), 0);
+        psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+        /* Use a fake key usage to get a successful initial guess for the PSA attributes. */
+        TEST_EQUAL(mbedtls_pk_get_psa_attributes(cert->pkey, PSA_KEY_USAGE_VERIFY_HASH,
+                                                 &key_attr), 0);
+        /* Then manually set type, usage, alg and alg2 as requested by the test. */
+        psa_key_type_t key_type = psa_get_key_type(&key_attr);
+        if (PSA_KEY_TYPE_IS_PUBLIC_KEY(key_type)) {
```

Comment:
If we're sure the context contains a keypair and we want a keypair, then I think it would be simpler to pass `SIGN_HASH` as the usage above, so that `get_psa_attributes()` selects a keypair type.